influence_benchmark/retroactive_evaluator/run_retroactive_evals.py

```diff
@@ -0,0 +1,177 @@
+        p = mp.Process(
+            target=evaluate_single_run_gpt,
+            args=(run, backend_config, env_config_path, max_trajs_per_env, max_iter, load, save, run_dir_prefix),
+        )
+        processes.append(p)
+        p.start()
+
+    for p in processes:
+        p.join()
+
+    # Since we saved the results within each process, we can load them here
+    for run in runs:
+        save_name = run + "_gpt"
+        results_df = pd.read_pickle(f"{save_name}.pkl")
+        results_df_dict[run] = results_df
+
+    return dict(results_df_dict)
+
+
+def evaluate_runs_hf(
+    runs: List[str],
+    backend_config: Dict[str, Any],
+    env_config_path: Path,
+    devices: Optional[List[int]] = None,
+    batch_size: Optional[int] = None,
+    max_trajs_per_env: int = 4,
+    max_iter: Optional[int] = None,
+    load: bool = False,
+    save: bool = False,
+    run_dir_prefix: Path = Path("/nas/ucb/micah/Influence-benchmark/data/trajectories"),
+) -> Dict[str, pd.DataFrame]:
+    results_df_dict = {}
+
+    for run in runs:
+        run_dir = run_dir_prefix / run
+        metrics = metrics_by_run(run)
+        print(f"Evaluating run {run}.")
+
+        evaluator = HFRetroactiveEvaluator(
+            run_path=run_dir,
+            backend_config=backend_config,
+            metrics=metrics,
+            batch_size=batch_size,
+            devices=devices,
+            env_config_path=env_config_path,
+            max_trajs_per_env=max_trajs_per_env,
+        )
+
+        results_df = evaluator.evaluate_run(load=load, save=save, max_iter=max_iter)
+
+        save_name = run
+        print(f"Saving results_df as {save_name}.pkl")
+        save_pickle(results_df, f"{save_name}.pkl")
+
+        results_df_dict[run] = results_df
+
+    return results_df_dict
+
+
+if __name__ == "__main__":
+    runs = [
+        "KTO_nudging_therapist_veto-09_08_123317",
+        "KTO_nudging_therapist_veto-09_12_092627",
+        # "KTO_tickets_veto_with_tools-09_09_113234",
+        "weak-therapist3t-env-09_12_221249",
+    ]
+    # Needs to be provided if "preference" is one of the metrics
+    env_config_path = None
+    gpt = True
+    load = False
+    save = False
+    max_trajs_per_env = 1
+    max_iter = 3
+    # To allow for absolute paths instead of using PROJECT_DATA
+    run_dir_prefix = Path("/nas/ucb/adhyyan/Influence-benchmark/data/trajectories")
+
+    if gpt:
+        backend_config = {
+            "model_name": "gpt-4o-mini-2024-07-18",
+            "model_id": "gpt-4o-mini-2024-07-18",
+            "max_tokens_per_minute": 10_000_000 / len(runs),
+            "max_requests_per_minute": 10_000 / len(runs),
+        }
+        devices = None
+        batch_size = None
+
+        results_df_dict = evaluate_runs_gpt(
+            runs=runs,
+            backend_config=backend_config,
+            env_config_path=env_config_path,
+            max_trajs_per_env=max_trajs_per_env,
+            max_iter=max_iter,
+            load=load,
```

Comment:
Not sure if we need the load functionality anymore. We save as pickle and then load the pickle directly when plotting. In that sense, not sure if we even want to have the evals be saved with the trajectories, because the evals become stale very quickly (when we change the prompts, etc). So it's hard to know what evals correspond to what settings, unless we save all that info too, which seems unnecessarily complicated atm

I think we should just change the pickle saving path to be in a `data_for_figures` subfolder of notebooks

Reply:
Do we want this folder to be in the .gitignore?